challenges/frontend/directions.md

```diff
@@ -0,0 +1,94 @@
+# Frontend Challenge
+
+## Summary
+
+Create a fictitious website using the Next.js framework and the designs provided by the client. 
+
+<img src="./screenshots/homepage.png" height="400" />
+<img src="./screenshots/homepage-with-dropdown.png" height="400" />
+<img src="./screenshots/mobile-version.png" height="400" />
+
+## Rationale
+We would like to evaluate your skill in the following ares: 
+- Taking existing designs and translating them to fully functional web presences. 
+- Writing production level code for the client and server
+- Communicating with team when working on the challenge 
+- Handling feedback from Teleport employees
+
+We believe this technique is not only better, but also is more fun compared to whiteboard/quiz interviews so common in the industry. It’s not without the downsides - it could take longer than traditional interviews. [Some of the best teams use coding challenges](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/). We appreciate your time and are looking forward to hack on this project together.
+
+## Tools
+- Backend - Node.js
+- Frontend - Typescript and React, Style Components
+- Framework - Next.js
+- Version Control - Github
+
+## Requirements
+There are 6 engineering levels at Teleport. It’s possible to score on level 1-5 through coding challenges. Level 6 is only for internal promotions. Check the engineering levels document for more details.
+
+Split the submission in 2-3 pull requests (one for each level of the test) for us to review. We will review every pull request and provide our feedback.
+
+# Directions
+
+You can find images for the challenge in **assets** folder along with the **Sketch** file used to create the design. 
+
+## Level 1 - Next.js setup with typescript and basic page layout
+- Create Next.js instance
+- Use typescript
+- Create React components for sections of the webpage, you do not need to style these components but the markup should be well thought out. 
+- Use yarn to setup the build process 
+- Provide documentation on how to get it running locally 
+
+## Level 2 - Create styles for these components
+- Use react style components to style the existing components you have created
+- Add functionality for dropdown menu
+- Add functionality for code box widget 
+- Add functionality for quote slider (you can use fake text for quotes and avatars) 
+
+## Level 3-4 - Responsive styling and unit testing, server side rendering
+- Provide unit tests for at least 2 of your components and at least one server side unit test. 
+- Migrate the application to the server so pages can be rendered on each http request from the server. 
+- Update all components so they work with any screen size or resolution 
```

Comment:
For level 3-4 I would expect someone who will be able to develop a fully featured Web Application, asking to write a few unit-tests seems and create responsive css looks more as level 1-2 requirements.

Reply:
@alex-kovoy @alexwolfe let's change the title from a frontend engineer to frontend-engineer-website to clarify the difference in both positions, because those are slightly different requirements.